go/dhcp/utils.go

```diff
@@ -0,0 +1,335 @@
+			if IP.Equal(VIPIp[v]) {
+				found = true
+				if VIP[v] == false {
+					fmt.Println(v + " got the VIP")
+					if _, ok := ControlIn[v]; ok {
+						Request := ApiReq{Req: "initialease", NetInterface: v, NetWork: ""}
+						ControlIn[v] <- Request
+					}
+					VIP[v] = true
+				}
+			}
+		}
+		if found == false {
+			VIP[v] = false
+		}
+	}
+}
+
+func NodeInformation(target net.HardwareAddr) (r NodeInfo) {
+
+	rows, err := database.Query("SELECT mac, status, IF(ISNULL(nc.name), '', nc.name) as category FROM node LEFT JOIN node_category as nc on node.category_id = nc.category_id WHERE mac = ?", target.String())
+	defer rows.Close()
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var (
+		Category string
+		Status   string
+		Mac      string
+	)
+	// Set default values
+	var Node = NodeInfo{Mac: target.String(), Status: "unreg", Category: "default"}
+
+	for rows.Next() {
+		err := rows.Scan(&Mac, &Status, &Category)
+		if err != nil {
+			log.Print(err)
+
+		}
+	}
+
+	Node = NodeInfo{Mac: Mac, Status: Status, Category: Category}
+	return Node
+}
+
+func ShuffleDNS(ConfNet pfconfigdriver.RessourseNetworkConf) (r []byte) {
+	if ConfNet.ClusterIPs != "" {
+		return Shuffle(ConfNet.ClusterIPs)
+	}
+	// if ConfNet.SplitNetwork == "enabled" {
+	// 	return Shuffle(ConfNet.Dns)
+	// 	// return []byte(net.ParseIP(ConfNet.Dns).To4())
+	// }
+	if ConfNet.Dnsvip != "" {
+		return []byte(net.ParseIP(ConfNet.Dnsvip).To4())
+	} else {
+		return []byte(net.ParseIP(ConfNet.Dns).To4())
+	}
+}
+
+func ShuffleGateway(ConfNet pfconfigdriver.RessourseNetworkConf) (r []byte) {
+	if ConfNet.NextHop != "" {
+		return []byte(net.ParseIP(ConfNet.NextHop).To4())
+	} else if ConfNet.ClusterIPs != "" {
+		return Shuffle(ConfNet.ClusterIPs)
+	} else {
+		return []byte(net.ParseIP(ConfNet.Gateway).To4())
+	}
+}
+
+func Shuffle(addresses string) (r []byte) {
+	var array []net.IP
+	for _, adresse := range strings.Split(addresses, ",") {
+		array = append(array, net.ParseIP(adresse).To4())
+	}
+
+	slice := make([]byte, 0, len(array))
+
+	random := rand.New(rand.NewSource(time.Now().UnixNano()))
+	for i := len(array) - 1; i > 0; i-- {
+		j := random.Intn(i + 1)
+		array[i], array[j] = array[j], array[i]
+	}
+	for _, element := range array {
+		elem := []byte(element)
+		slice = append(slice, elem...)
+	}
+	return slice
+}
+
+func ShuffleNetIP(array []net.IP) (r []byte) {
+
+	slice := make([]byte, 0, len(array))
+
+	random := rand.New(rand.NewSource(time.Now().UnixNano()))
+	for i := len(array) - 1; i > 0; i-- {
+		j := random.Intn(i + 1)
+		array[i], array[j] = array[j], array[i]
+	}
+	for _, element := range array {
+		elem := []byte(element)
+		slice = append(slice, elem...)
+	}
+	return slice
+}
+
+func ShuffleIP(a []byte) (r []byte) {
+
+	var array []net.IP
+	for len(a) != 0 {
+		array = append(array, net.IPv4(a[0], a[1], a[2], a[3]).To4())
+		_, a = a[0], a[4:]
+	}
+	return ShuffleNetIP(array)
+}
+
+func ByteToString(a []byte) string {
```

Comment:
I guess this does something different than string([]byte{192,168,5,3})

Could you add some godoc for this and move it to sharedutils if this could be useful somewhere else